raocp/core/scenario_tree.py

```diff
@@ -106,22 +114,39 @@ def __make_ancestors_values_stages(self):
                 node_id = cursor + i
                 cover = self.__cover(int(values[node_id]))
                 length_cover = len(cover)
-                ancestors = np.concatenate((ancestors, node_id * np.ones((length_cover, ))))
+                ones = np.ones((length_cover, ), dtype=int)
+                ancestors = np.concatenate((ancestors, node_id * ones))
                 nodes_added_at_stage += length_cover
                 values = np.concatenate((values, cover))
             num_nodes_at_stage = nodes_added_at_stage
             cursor = cursor_new
-            stages = np.concatenate((stages, (1 + stage_idx) * np.ones(nodes_added_at_stage, )))
+            ones = np.ones(nodes_added_at_stage, dtype=int)
+            stages = np.concatenate((stages, (1 + stage_idx) * ones))
 
         for stage_idx in range(self.__stopping_time, self.__num_stages):
             ancestors = np.concatenate((ancestors, range(cursor, cursor+num_nodes_at_stage)))
             cursor += num_nodes_at_stage
-            stages = np.concatenate((stages, (1 + stage_idx) * np.ones(nodes_added_at_stage, )))
-            # TODO update `values` after stopping time?
+            ones = np.ones((nodes_added_at_stage,), dtype=int )
+            stages = np.concatenate((stages, (1 + stage_idx) * ones))
+            values = np.concatenate((values, values[-num_nodes_at_stage::]))
+
         return ancestors, values, stages
 
     def __make_probability_values(self):
-        return 0
+        """
+        :return: probability
+        """
+        num_nonzero_init_distr = len(list(filter(lambda x: (x > 0), self.__initial_distribution)))
+        ancestors, values, stages = self.__make_ancestors_values_stages()
```

Comment:
`self.__make_ancestors_values_stages` is a time consuming method that has been already invoked; you don't need to call it again